client-sdk/go/modules/contracts/contracts.go

```diff
@@ -222,6 +228,43 @@ func (a *v1) Custom(ctx context.Context, round uint64, id InstanceID, data, rsp
 	return nil
 }
 
+// Implements V1.
+func (a *v1) GetEvents(ctx context.Context, instanceID InstanceID, round uint64) ([]*Event, error) {
+	rawEvs, err := a.rc.GetEventsRaw(ctx, round)
+	if err != nil {
+		return nil, err
+	}
+
+	evs := make([]*Event, 0)
+	for _, rawEv := range rawEvs {
+		ev, err := a.DecodeEvent(rawEv)
+		if err != nil {
+			return nil, err
+		}
+		if ev == nil {
+			continue
+		}
+		if ev.(*Event).ID != instanceID {
+			continue
+		}
+		evs = append(evs, ev.(*Event))
+	}
+
+	return evs, nil
+}
+
+// Implements client.EventDecoder.
+func (a *v1) DecodeEvent(event *types.Event) (client.DecodedEvent, error) {
+	if !strings.HasPrefix(event.Module, ModuleName) {
```

Comment:
Probably should be that either it is `ModuleName` or the prefix is `ModuleName + "."`. So for example if there is a module called `contractsfoo` it doesn't clash.